src/Microsoft.AspNetCore.Authentication.Abstractions/AuthenticationScheme.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    public class AuthenticationScheme
+    {
+        public AuthenticationScheme(string name, Type handlerType, IReadOnlyDictionary<string, object> settings)
+        {
+            Name = name ?? throw new ArgumentNullException(nameof(name));
+            if (!typeof(IAuthenticationHandler).IsAssignableFrom(handlerType))
+            {
+                throw new ArgumentException("handlerType must implement IAuthenticationSchemeHandler.");
+            }
+            HandlerType = handlerType ?? throw new ArgumentNullException(nameof(handlerType));
+            Settings = settings ?? throw new ArgumentNullException(nameof(settings));
+        }
+
+        // TODO: add display name?
+        public string Name { get; }
+        public Type HandlerType { get; }
+
+        // Holds things like the configured options instances for the handler
+        // Also replacement for AuthenticationDescription
+        public IReadOnlyDictionary<string, object> Settings { get; }
```

Comment:
Yeah this can go away now, since we can add/remove options dynamically I don't think this is needed at all.  Fleshing out the add/remove auth scheme using custom options in the next update